code/game/area/areas.dm

```diff
@@ -307,12 +309,24 @@ var/global/list/mob/living/forced_ambiance_list = new
 
 	play_ambience(L)
 	L.lastarea = newarea
+	do_area_blurb(L)
 
 /area/Exited(A)
 	if(isliving(A))
 		clear_ambience(A)
 	return ..()
 
+/area/proc/do_area_blurb(var/mob/living/L)
+	if(isnull(description))
+		return
+
+	if(!(L && L.client && L.get_preference_value(/datum/client_preference/area_info_blurb) == PREF_YES))
+		return 
+	
+	if(!(L.client in blurbed_stated_to))
+		blurbed_stated_to += L.client
```

Comment:
Yeah, don't shove client refs into long lived lists please. Clients that log out and back in will be shown the message again.

Add the ckey instead.

Reply:
Addressed.